packages/skeleton-react/src/lib/components/Rating/Rating.test.tsx

```diff
@@ -0,0 +1,54 @@
+import { describe, expect, it } from 'vitest';
+import { render } from '@testing-library/react';
+
+import { Rating } from './Rating.js';
+
+describe('<Rating>', () => {
+	it('should render the component', () => {
+		const { getByTestId } = render(<Rating />);
+		const component = getByTestId('rating');
+		expect(component).toBeInTheDocument();
+	});
+
+	it('should render with a value', () => {
+		const testValue = 14;
+		const { getByTestId } = render(<Rating value={testValue} />);
+		const input = getByTestId('rating-input');
+		expect(input.getAttribute('value')).eq(`${testValue}`);
+	});
+
+	it('should render with custom iconEmpty', () => {
+		const testIconEmpty = 'testIconEmpty';
+		const { getAllByTestId } = render(<Rating value={0} iconEmpty={testIconEmpty} />);
+		const elementIcons = getAllByTestId('rating-item');
+		expect(elementIcons[0]).toHaveTextContent(testIconEmpty);
+	});
+
+	it('should render with custom iconHalf', () => {
+		const testIconHalf = 'testIconHalf';
+		const { getAllByTestId } = render(<Rating value={0.5} iconHalf={testIconHalf} allowHalf />);
+		const elementIcons = getAllByTestId('rating-item');
+		expect(elementIcons[0]).toHaveTextContent(testIconHalf);
+	});
+
+	it('should render with custom iconFull', () => {
+		const testIconFull = 'testIconFull';
+		const { getAllByTestId } = render(<Rating value={1} iconFull={testIconFull} />);
+		const elementIcons = getAllByTestId('rating-item');
+		expect(elementIcons[0]).toHaveTextContent(testIconFull);
+	});
+
+	it('should render base classes on root', () => {
```

Comment:
I was thinking about that, but I'm not sure if we need to test every single style prop. We've never really had an issue with props being removed or changed, as that typically falls into the category of a major change. Testing base/classes is a nice litmus though because these are present on nearly every element that take style props.

Reply:
Makes sense, although doing it through a loop is simply removing or adding items to the array so I'll keep mine but we can add them when the component is finalized.

Reply:
Agreed. I think that's the right approach. We'll plan to revisit the test cases prior to launch when things are solidified.